Lib/test/support/__init__.py

```diff
@@ -2725,7 +2725,7 @@ def missing_compiler_executable(cmd_names=[]):
         if cmd_names:
             assert cmd is not None, \
                     "the '%s' executable is not configured" % name
-        elif cmd is None:
+        elif not cmd or cmd is None:
```

Comment:
`not cmd` shadows `cmd is None` test

Reply:
do you mean, this should be "elif (not cmd) or (cmd is None):

What suprised me is is that "not cmd" and "cmd is None" are different (being an old C debugger I was reading cmd is None to be "cmd == 0".

In short, as (not cmd) != (cmd is None) line 2730 fails - when (not cmd) == True, cmd[0] generates an error.

Reply:
OK, did not add comment when I marked "resolved".

"Resolved" is as much as I am making this its own issue (or finding an existing one that fits) - and will post a new PR - so that the PR that remains makes use of other comments re: self._isgcc